### PR TITLE
Added output variable support for DockerV0  and DockerV1 tasks

### PR DESCRIPTION
**Pull request by Anumita**
Added support to solve #4005

### CODE REVIEW

File: Tasks/DockerV0/container.ts [hunk 1]
```diff
@@ -49,10 +51,11 @@ require({
...
     "Run a Docker command": "./containercommand"
-}[action]).run(connection)
+}[action]).run(connection, (data) => result += data)
 /* tslint:enable:no-var-requires */
 .fin(function cleanup() {
     connection.close();
+    tl.setVariable("DockerOutput", result);
 })
 .then(function success() {
     tl.setResult(tl.TaskResult.Succeeded, "");
```
Comment (bansalaseem) [inline Tasks/DockerV0/container.ts]: should we set the output only on success?

File: Tasks/DockerV0/container.ts [hunk 2]
```diff
@@ -49,10 +51,11 @@ require({
     "Push images": "./containerpush",
     "Run an image": "./containerrun",
     "Run a Docker command": "./containercommand"
-}[action]).run(connection)
+}[action]).run(connection, (data) => result += data)
 /* tslint:enable:no-var-requires */
 .fin(function cleanup() {
     connection.close();
+    tl.setVariable("DockerOutput", result);
```
Comment (bansalaseem) [inline Tasks/DockerV0/container.ts]: this also should be done only on success

File: Tasks/DockerV0/container.ts (hunk 1 above)
Comment (tvikrantht) [reply Tasks/DockerV0/container.ts]: +1

**Comment by devedse** [comment]
@Anumita , is this already released? I'm trying to basically create a VSTS build that gets all images and then cleans them up, so basically
1. Docker task with `docker images -q` that stores it in a variable output
2. Docker task with `docker rmi $(output)`

However I run into the following error then:
image

```
2018-10-11T20:10:28.3414397Z [command]/usr/bin/docker -H tcp://******************.westeurope.cloudapp.azure.com:2376 --tls --tlscacert='.dockercerts/ca.pem' --tlscert='.dockercerts/cert.pem' --tlskey='.dockercerts/key.pem' rmi $(output)
2018-10-11T20:10:28.3414640Z Error response from daemon: invalid reference format
2018-10-11T20:10:28.3469892Z ##[error]Error response from daemon: invalid reference format
2018-10-11T20:10:28.3480240Z ##[error]/usr/bin/docker failed with return code: 1
```

**Comment by Anumita** [comment]
@devedse ,  it will take about 3 weeks to reach your account as the change is in sprint 142 payload.

**Pull request merged**